src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -225,7 +223,7 @@ std::error_code partition_allocator::check_cluster_limits(
         const uint64_t memory_limit = effective_cluster_memory
                                       / memory_per_partition_replica.value();
 
-        if (memory_limit > 0 && proposed_total_partitions > memory_limit) {
```

Comment:
This check is removed so that if we have no info we allow everything?

Reply:
The idea was that `memory_limit` should never be '0' in the happy path now. My assumption was that it would only be '0' when running on redpanda < 21 and that we can ignore that version in this code. Now, we should get a value from `available_memory_bytes` if available but definitely `available_memory_gb` if not.

The only other situation `memory_limit` could be 0 is it something went catastrophically wrong with the effective cluster memory calculation in which case I think we should trigger this warning.